src/main/kotlin/org/usfirst/frc/team4099/robot/subsystems/Elevator.kt

```diff
@@ -0,0 +1,243 @@
+package org.usfirst.frc.team4099.robot.subsystems
+
+import com.ctre.phoenix.motorcontrol.ControlMode
+import com.ctre.phoenix.motorcontrol.FeedbackDevice
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard
+import org.usfirst.frc.team4099.lib.util.CANMotorControllerFactory
+import org.usfirst.frc.team4099.lib.util.Utils
+import org.usfirst.frc.team4099.lib.util.conversions.ElevatorConversion
+import org.usfirst.frc.team4099.robot.Constants
+import org.usfirst.frc.team4099.robot.loops.Loop
+
+class Elevator private constructor(): Subsystem {
+    private val talon = CANMotorControllerFactory.createDefaultTalon(Constants.Elevator.ELEVATOR_TALON_ID)
+
+    private var elevatorPower = 0.0
+    var elevatorState = ElevatorState.OPEN_LOOP
+    var movementState = MovementState.STILL
+        private set
+    var observedElevatorPosition = 0.0
+        private set
+    var observedElevatorVelocity = 0.0
+        private set
+    var isHatchPanel = true
+
+    enum class ElevatorState (val targetPos : Double) {
```

Comment:
Idk if you want to add this, but perhaps have a state when the elevator is completely down, as we may not want to have the elevator raised at all during certain times (ex. when climbing)